backend/src/hooks/check-has-permission/check-has-permission.hook.ts

```diff
@@ -21,16 +23,36 @@ const checkHasPermissions =
       });
     }
 
-    const hasUserPermission = checkHasPermission({
-      permissionKeys: pagePermissions,
-      userPermissions: user.permissions,
-    });
+    switch (type) {
+      case 'every': {
+        const hasUserAllPermissions = checkHasPermission({
```

Comment:
Please take a look what's going on in shared `checkHasPermission` helper - currently that helper uses `.some` on permissions check - so seems like your changes with 'every'/'oneOf' won't make an effect

Reply:
Maybe it would make sense to enhance shared `checkHasPermission` with same `checkType` - but make it optional with default value equal to current implementation, and pass specific type by occasion (in BE hook always)